src/esp/sensor/Base.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include "esp/io/json.h"
+#include "esp/scene/SceneNode.h"
+
+#include <string>
+
+namespace esp {
+namespace sensor {
+class BaseSensor : public Magnum::SceneGraph::AbstractFeature3D {
+ public:
+  explicit BaseSensor(scene::SceneNode& node, const esp::io::JsonDocument spec);
+  std::string getType();
+  std::string getUUID();
+  std::vector<char> getObservation();
```

Comment:
Agree with @erikwijmans but I think this might even be better wrapped into an `Observation` type which we can then endow with additional information about the observation down the line.  There's a stub for this that hasn't been hooked up yet at https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/sensor/Sensor.h#L67-L71 .  Perhaps we should adjust that to contain `std::vector<char>` and use it here?

Reply:
Agree with @msavva. Returning an Observation class instance makes sense. One piece of data I think of that might be useful is timestamp. An Observation class would also be useful for handling GPU copy.

Reply:
Thanks. Let's target that in a separate incremental PR. I will add a TODO here.